doc/source/dev/conduct/code_of_conduct.rst

```diff
@@ -0,0 +1,164 @@
+understanding of a collaborative, shared environment and goals. We expect it to
+be followed in spirit as much as in the letter, so that it can enrich all of us
+and the technical community in which we participate.
+
+
+Specific Guidelines
+-------------------
+
+We strive to:
+
+1. Be open. We invite anyone to participate in our community. We preferably use
+   public methods of communication for project-related messages, unless
+   discussing something sensitive. This applies to messages for help or
+   project-related support, too; not only is a public support request much more
+   likely to result in an answer to a question, it also makes sure that any
+   inadvertent mistakes made by people answering will be more easily detected
+   and corrected.
+
+2. Be empathetic, welcoming, friendly, and patient. We work together to resolve
+   conflict, assume good intentions, and do our best to act in an empathetic
+   fashion. We may all experience some frustration from time to time, but we do
+   not allow frustration to turn into a personal attack. A community where
+   people feel uncomfortable or threatened is not a productive one. We should
+   be respectful when dealing with other community members as well as with
+   people outside our community.
+
+3. Be collaborative. Our work will be used by other people, and in turn we will
+   depend on the work of others. When we make something for the benefit of the
+   project, we are willing to explain to others how it works, so that they can
+   build on the work to make it even better. Any decision we make will affect
+   users and colleagues, and we take those consequences seriously when making
+   decisions.
+
+3. Be inquisitive. Nobody knows everything! Asking questions early avoids many
+   problems later, so questions are encouraged, though they may be directed to
+   the appropriate forum. Those who are asked should be responsive and helpful,
+   within the context of our shared goal of improving SciPy project code,
+   documentation and other content.
+
+4. Be careful in the words that we choose. Whether we are participating as
+   professionals or volunteers, we value professionalism in all interactions,
+   and take responsibility for our own speech. Be kind to others. Do not insult
+   or put down other participants. Harassment and other exclusionary behaviour
+   are not acceptable. This includes, but is not limited to:
+
+    - Violent threats or language directed against another person.
+    - Sexist, racist, or otherwise discriminatory jokes and language.
+    - Posting sexually explicit or violent material.
+    - Posting (or threatening to post) other people's personally identifying information ("doxing").
+    - Sharing private content, such as emails sent privately or non-publicly, or unlogged forums such as IRC channel history.
+    - Personal insults, especially those using racist or sexist terms.
+    - Unwelcome sexual attention.
+    - Excessive or unnecessary profanity.
```

Comment:
Perhaps modify to "Excessive profanity."; because unnecessary is a much more debated issue.  Some people feel phrases like "shitty algorithm" is unnecessarily profane.